src/main/java/com/michelin/ns4kafka/controller/SchemaController.java

```diff
@@ -160,24 +162,59 @@ public Mono<HttpResponse<Void>> delete(String namespace, @PathVariable String su
             return Mono.error(new ResourceValidationException(SCHEMA, subject, invalidOwner(subject)));
         }
 
-        return schemaService.getLatestSubject(ns, subject)
+        // delete all versions of the schema
+        if (version.isEmpty()) {
```

Comment:
OK for this `if-else` logic. It looks it can be simplified in a single expression like:

```java
return version
  .<Mono<HttpResponse<Void>>>map(v -> schemaService.getSubject(ns, subject, v)
    ...
  .orElseGet(() -> schemaService.getLatestSubject(ns, subject)
    ...
  );
```

Reply:
The actions performed between the two cases are not the same:

1. In the first case `version.isEmpty()` (need to delete all versions), we just call the `deleteAllVersions` method to delete all the versions of the schema.
2. In the second case (need to delete a specific version), we call the `deleteVersion` method to delete a specific version, and depending on the situation, we need to send different event log:
- If we deleted the **last** version of the schema, we send a `deleted` status. (no schema is remaining)
- If we deleted the **latest** version but there is still one remaining version, we send a `changed` status.
- If we deleted the non-latest version, we send no event log. (the latest version was not deleted and remains the latest version)

Reply:
Agree with the whole logic, I was just suggesting to update the `if-else` condition with the `Optional#orElseGet` logic

Reply:
By changing the sentEventLog to sending only `deleted` status, I setup version.map().orElseGet()